### DISCUSSION TITLE
iOS integration with Compose Multiplatform giving error

### DISCUSSION THREAD

**Discussion by niranjk** [GitLiveApp/firebase-kotlin-sdk#431] [Q&A] [answered]
I got this error while implementing: 
  **implementation("dev.gitlive:firebase-firestore:1.10.4") // This line
  implementation("dev.gitlive:firebase-common:1.10.4")// This line**

_**ld: Undefined symbols:
  _FIRFirestoreErrorDomain, referenced from:
      _cocoapods_FirebaseFirestore_FIRFirestoreErrorDomain_getter_wrapper0 in ComposeApp[arm64][4](libdev.gitlive:firebase-firestore-cinterop-FirebaseFirestore-cache.a.o)
  _OBJC_CLASS_$_FIRDocumentSnapshot, referenced from:
       in ComposeApp[arm64][2630](libdev.gitlive:firebase-firestore-cache.a.o)
  _OBJC_CLASS_$_FIRFirestore, referenced from:
       in ComposeApp[arm64][2630](libdev.gitlive:firebase-firestore-cache.a.o)
  _OBJC_CLASS_$_FIRQuerySnapshot, referenced from:
       in ComposeApp[arm64][2630](libdev.gitlive:firebase-firestore-cache.a.o)
clang: error: linker command failed with exit code 1 (use -v to see invocation)**_

**Comment by mr-kew** [comment]
Are you using CocoaPods? If so, is the framework linked statically or dynamically?

This seems like exactly the same error I get, when I use isStatic = false on CocoaPods framework. Linking it statically should fix the problem, but then XCode Previews wont work. I have not found a full solution yet, but I have it currently [posted on SO](https://stackoverflow.com/questions/77616038/how-to-get-xcode-preview-working-in-kmp-with-firebasefirestore-using-cocoapods).

**Reply by niranjk** [reply, marked as answer]
Thanks ;) I already solved it: https://medium.com/@niranjanky14/i-tried-integrating-firebase-in-the-compose-multiplatform-check-out-what-i-found-out-0f3b8d3c84cb

**Reply by mr-kew** [reply]
@niranjk  Nice, did you get even the previews working?

**Reply by niranjk** [reply]
Yes, working fine. Thanks.
